documentation/design-docs/diagnostics-client-library.md

```diff
@@ -0,0 +1,327 @@
+        /// <param name="circularBufferMB">The size of the runtime's buffer for collecting events in MB</param>
+        /// <param name="requestRundown">If true, request rundown events from the runtime</param>
+        /// <returns>
+        /// An EventPipeSession object representing the EventPipe session that just started.
+        /// </returns> 
+        public static EventPipeSession StartTracing(int processId, IEnumerable<EventPipeProvider> providers, int circularBufferMB=256, bool requestRundown=true)
+
+        /// <summary>
+        /// Stop the EventPipe session provided as an argument.
+        /// </summary>
+        /// <param name="session">The EventPipeSession to be stopped.
+        /// <returns>
+        /// true if stopping the tracing session succeeded. false otherwise. 
+        /// </returns> 
+        public static bool StopTracing(EventPipeSession session)
+
+        /// <summary>
+        /// Trigger a core dump generation.
+        /// </summary> 
+        /// <param name="processId">Target process' ID</param>
+        /// <param name="dumpName">Name of the dump to be generated</param>
+        /// <param name="dumpType">Type of the dump to be generated</param>
+        /// <param name="logDumpGeneration">When set to true, display the dump generation debug log to the console</param>
+        /// <returns>
+        /// true if stopping the tracing session succeeded. false otherwise. 
+        /// </returns> 
+        public static GenerateCoreDump(int processId, string dumpName, DumpType dumpType, bool logDumpGeneration)
+
+        /// <summary>
+        /// Attach a profiler.
+        /// </summary>
+        /// <param name="processId">Target process' ID</param>
+        /// <param name="attachTimeout">Timeout in seconds</param>
+        /// <param name="attachTimeout">
+        public static int AttachProfiler(int processId, uint attachTimeout, Guid profilerGuid, string profilerPath, byte[] additionalData);
+
+    }
+}
+```
+
+
+### Exceptions that can be thrown 
+
+```cs
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    // Generic wrapper for exceptions thrown by this library
+    public class DiagnosticsClientExcpetion : Exception {} 
+
+    // When a certian command is not supported by either the library or the target process' runtime
+    public class UnsupportedProtocolException : DiagnosticsClientException {} 
+
+    // When the runtime is no longer availble for attaching.
+    public class ServerNotAvailableException : DiagnosticsClientException {} 
+
+    // When the runtime responded with an error
+    public class ServerThrownException : DiagnosticsClientException {} 
+}
+```
+
+## Micorosft.Diagnostics.Client.EventPipe namespace
+
+### EventPipeProvider
+A class that describes an EventPipe provider.
+```cs
+namespace Microsoft.Diagnostics.Client
+{
+    public class EventPipeProvider
+    {
+        public EventPipeProvider(
+            string name,
+            long keywords = ulong.MaxValue,
```

Comment:
You mentioned these were the 'current' default... did you mean current in RuntimeClient source or elsewhere? From the API design perspective I give RuntimeClient's current source low weight, TraceEvent medium weight, and EventListener the highest weight because stuff that goes into BCL usually has the most review. EventListener uses EventKeywords.None as the default keyword value when not specified and doesn't provide a default level. That sounds pretty reasonable to me.

Reply:
I changed it to match EventListener.